c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1362,6 +1365,9 @@ class DeviceCachingAllocator {
       return;
     }
     block->stream_uses.insert(stream);
+    if (C10_UNLIKELY(!captures_underway.empty())) {
```

Comment:
the cudagraph_stream_uses should only be present during capture.. wouldn't ever make sense for us to checkpoint it after capture is done. another reason to make it a map instead of per-block property